html/webappapis/dynamic-markup-insertion/opening-the-input-stream/quirks.window.js

```diff
@@ -0,0 +1,69 @@
+test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  t.add_cleanup(() => frame.contentDocument.close());
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+  frame.contentDocument.open();
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.close();
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+}, "document.open() and quirks mode (iframe, no doctype)");
+
+test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  t.add_cleanup(() => frame.contentDocument.close());
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+  frame.contentDocument.open();
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.write("<!doctype html public");
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.write(" \"-//IETF//DTD HTML 3//\"");
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.write(">");
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+  frame.contentDocument.close();
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+}, "document.open() and quirks mode (iframe, old doctype)");
+
+test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  t.add_cleanup(() => frame.contentDocument.close());
+  assert_equals(frame.contentDocument.compatMode, "BackCompat");
+  frame.contentDocument.open();
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.write("<!doctype html");
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.write(">");
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+  frame.contentDocument.close();
+  assert_equals(frame.contentDocument.compatMode, "CSS1Compat");
+}, "document.open() and quirks mode (iframe, new doctype)");
+
+// Test derived from quirks/blocks-ignore-line-height.html.
```

Comment:
There's enough here to figure out that the test isn't really about the layout size of those divs, but can you spell out that this is all because there's no way to directly test if one is in limited quirks mode, and that this test is only valid on the assumption that the one linked here passes?

Reply:
Done.